mlos_bench/mlos_bench/services/remote/azure/azure_services.py

```diff
@@ -81,6 +83,17 @@ class AzureVMService(Service, SupportsVMOps, SupportsRemoteExec):
         "?api-version=2022-03-01"
     )
 
+    # From: https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/deallocate
+    _URL_DEPROVISION = (
+        "https://management.azure.com" +
+        "/subscriptions/{subscription}" +
+        "/resourceGroups/{resource_group}" +
+        "/providers/Microsoft.Compute" +
+        "/virtualMachines/{vm_name}" +
+        "/delete" +
+        "?api-version=2022-03-01"
+    )
```

Comment:
This is maybe something to move to a separate PR, but we'd previously been too loose with our terminology - VM deprovisioning actually deletes the VM, whereas deallocation releases the compute resources associated with that VM without actually deleting the VM object (e.g., it's disks).
See Also #468 for related commentary about cleaning up after ourselves for cost reduction reasons.

Reply:
yeah, let's do it in a separate PR. rn I am mostly concerned about keeping the framework usable after all our refactorings. I am fine with the changes as long as I can still run our existing configs at least for Redis and MySQL

Reply:
I left a comment there referencing a new issue ( #498 ) and restored the original URL for now.